File: sdks/typescript/src/resources/admin/connectors.ts
import type { RequestClient } from "../../core.js";
import type { GoogleConnectorConfig, UpdateGoogleConnectorConfigBody } from "../../types/index.js";

export class AdminConnectorsResource {
  readonly google: AdminGoogleConnectorResource;

  constructor(client: RequestClient) {
    this.google = new AdminGoogleConnectorResource(client);
  }
}

export class AdminGoogleConnectorResource {
  constructor(private readonly _client: RequestClient) {}

  get(): Promise<GoogleConnectorConfig> {
    return this._client._request("GET", "/v1/admin/connectors/google");
  }

  update(body: UpdateGoogleConnectorConfigBody): Promise<GoogleConnectorConfig> {
    return this._client._request("PUT", "/v1/admin/connectors/google", { json: body });
  }
}

File: sdks/python/src/bigrag/resources/admin/connectors.py
from __future__ import annotations

from typing import TYPE_CHECKING

from bigrag.types.connectors import (
    GoogleConnectorConfig,
    UpdateGoogleConnectorConfigBody,
)

if TYPE_CHECKING:
    from bigrag._core import BigRAGCore


class AdminConnectorsResource:
    google: AdminGoogleConnectorResource

    def __init__(self, client: BigRAGCore) -> None:
        self.google = AdminGoogleConnectorResource(client)


class AdminGoogleConnectorResource:
    def __init__(self, client: BigRAGCore) -> None:
        self._client = client

    async def get(self) -> GoogleConnectorConfig:
        return await self._client._request("GET", "/v1/admin/connectors/google")

    async def update(
        self, body: UpdateGoogleConnectorConfigBody
    ) -> GoogleConnectorConfig:
        return await self._client._request(
            "PUT", "/v1/admin/connectors/google", json=body
        )
